spec/dns_spec.rb

```diff
@@ -0,0 +1,111 @@
+describe LinuxAdmin::Dns do
+  SEARCH_ORDER = %w(ewr.redhat.com ewr.redhat.com. redhat.com. manageiq.com)
+  NAMESERVERS  = %w(10.11.5.19 10.5.30.160 192.168.252.3)
+
+  NEW_ORDER = %w(thing.com other.com)
+  NEW_NS    = %w(1.2.3.4 5.6.7.8)
+
+  def dns_file_path
+    Pathname.new(data_file_path("dns/resolv.conf"))
+  end
+
+  let(:dns) { described_class.new(dns_file_path) }
+
+  around do |ex|
+    data = File.read(dns_file_path)
+    ex.run
+    File.write(dns_file_path, data)
+  end
+
+  describe ".new" do
+    it "sets the filename" do
+      expect(dns.filename).to eq(dns_file_path)
+    end
+
+    it "parses the nameservers" do
+      expect(dns.nameservers).to eq(NAMESERVERS)
+    end
+
+    it "parses the search order" do
+      expect(dns.search_order).to eq(SEARCH_ORDER)
+    end
+  end
+
+  describe "#reload" do
+    NEW_CONTENTS = <<DNS_END
+search thing.com other.com
+nameserver 1.2.3.4
+nameserver 5.6.7.8
+DNS_END
+
+    def change_file
+      File.write(dns_file_path, NEW_CONTENTS)
```

Comment:
We should try to avoid writing files in a spec, especially files that hold test data.  In the spec, you should be able to:
```ruby
allow(File).to receive(:read).and_return(original_content, new_content)
```